tools/build_api.py

```diff
@@ -190,16 +190,23 @@ def build_project(src_path, build_path, target, toolchain_name,
 
     try:
         # Scan src_path and libraries_paths for resources
-        resources = toolchain.scan_resources(src_paths[0])
-        for path in src_paths[1:]:
-            resources.add(toolchain.scan_resources(path))
+        resources = sum(map(toolchain.scan_resources,src_paths), None)
+        resources.objects.extend(extra_resources.objects)
+        resources.inc_dirs.extend(extra_resources.inc_dirs)
+        resources.headers.extend(extra_resources.headers)
+        resources.libraries.extend(extra_resources.libraries)
+        resources.hex_files.extend(extra_resources.hex_files)
+        resources.bin_files.extend(extra_resources.bin_files)
+        resources.json_files.extend(extra_resources.json_files)
```

Comment:
extra_resources() might contain features. This implementation doesn't take this into account. Also paths to original dirs are lost. See Resource __add__()